src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/PayRatesService.kt

```diff
@@ -89,7 +89,7 @@ class PayRatesService(
     }
 
   private fun CourseActivityPayRate.rateIsUnchanged(requested: PayRateRequest) =
-    this.halfDayRate.compareTo(requested.rate) == 0
+    this.halfDayRate.compareTo(CourseActivityPayRate.preciseHalfDayRate(requested.rate)) == 0
```

Comment:
Why not use .equals() ?

Reply:
Because that's what [Kotlin does under the hood](https://kotlinlang.org/docs/equality.html)

Reply:
Oh I see you mean why compareTo instead of equals! I think it was because equals doesn't work with BigDecimals